tests/generator-Tests/Unit-Tests/CodeGeneratorExpectedResults/JavaInterop1/WriteKotlinUnsignedArrayTypePropertiesClass.txt

```diff
@@ -0,0 +1,130 @@
+// Metadata.xml XPath class reference: path="/api/package[@name='java.code']/class[@name='MyClass']"
+[global::Android.Runtime.Register ("java/code/MyClass", DoNotGenerateAcw=true)]
+public partial class MyClass  {
+
+	static readonly JniPeerMembers _members = new JniPeerMembers ("java/code/MyClass", typeof (MyClass));
+	internal static new IntPtr class_ref {
+		get {
+			return _members.JniPeerType.PeerReference.Handle;
+		}
+	}
+
+	protected MyClass (IntPtr javaReference, JniHandleOwnership transfer) : base (javaReference, transfer) {}
+
+	public unsafe uint[] UIntProp {
```

Comment:
This is wrong: we should *never* map "property-looking" methods to actual properties when the return type is a property.  See e.g.

  * https://github.com/xamarin/java.interop/blob/d61b329cbdeaa6d42c17618de59cb850ab82ce93/tests/generator-Tests/expected/EnumerationFixup/Xamarin.Test.SomeObject2.cs#L116
  * https://github.com/xamarin/java.interop/blob/d61b329cbdeaa6d42c17618de59cb850ab82ce93/tools/generator/Java.Interop.Tools.Generator.ObjectModel/Method.cs#L77

Looks like unsigned types are being treated differently from signed types, such that `RetVal.IsArray` is returning false for this array type.

Reply:
Yeah, this test is actually irrelevant because I found out you cannot create a Kotlin property with an array type, but I definitely need to dig in and figure out what's going on with `IsArray` returning `false`. 👍

Reply:
Ah, `IsArray` is returning `true` like it should, this unit test just doesn't go through that logic.  It is explicitly creating properties with array types rather than combining methods into a property.

https://github.com/xamarin/java.interop/pull/539/files#diff-a75d7dad4d297b1451cc51ec2b8b2481R85